application-task-default/src/main/java/com/xwiki/task/internal/TaskBlockProcessor.java

```diff
@@ -77,10 +77,23 @@ public class TaskBlockProcessor
      * @return a {@link LinkBlock} that points to the reference and the number of the task as content.
      */
     public Block createTaskLinkBlock(String reference, int taskNumber)
+    {
+        return createTaskLinkBlock(reference, taskNumber, ResourceType.DOCUMENT);
+    }
+
+    /**
+     * Create a link block for a task.
+     *
+     * @param reference the reference to the task page.
+     * @param taskNumber the number of the task, that uniquely identifies it.
+     * @param resourceType the resource type of the reference.
+     * @return a {@link LinkBlock} that points to the reference and the number of the task as content.
+     */
+    public Block createTaskLinkBlock(String reference, int taskNumber, ResourceType resourceType)
```

Comment:
This new signature doesn't seem to be used, so do you really need to add it? Asking because this is a public API, where it's hard to remove stuff once added.

Reply:
I might not need it right now, but it will be needed later when we replace the usage of DocumentReference with PageReferences.